presto-hive/src/main/java/io/prestosql/plugin/hive/HiveClientModule.java

```diff
@@ -62,7 +62,8 @@ public void configure(Binder binder)
         binder.bind(TypeTranslator.class).toInstance(new HiveTypeTranslator());
         binder.bind(CoercionPolicy.class).to(HiveCoercionPolicy.class).in(Scopes.SINGLETON);
 
-        binder.bind(HdfsConfigurationUpdater.class).in(Scopes.SINGLETON);
+        binder.bind(HdfsConfigurationInitializer.class).in(Scopes.SINGLETON);
+        newSetBinder(binder, DynamicConfigurationProvider.class);
```

Comment:
I think that order in they are called matters, how are you going to make sure they are executed in correct order? Or how we can control this?

Reply:
They are not order dependent in any of the envisioned use cases. If they are, then the entire design of having them provided by different Guice models won't work. It's expected that they will work like the GCS one, where each one handles a different FS scheme.